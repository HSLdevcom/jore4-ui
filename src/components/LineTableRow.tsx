import React from 'react';
import { useTranslation } from 'react-i18next';
import { MdPinDrop } from 'react-icons/md';
import { Link } from 'react-router-dom';
import { RouteLine } from '../generated/graphql';
import { useShowRoutesOnModal } from '../hooks';
import { Column, Row } from '../layoutComponents';
import { Path, routes } from '../routes'; // eslint-disable-line import/no-cycle
import { mapToShortDate, MAX_DATE, MIN_DATE } from '../time';
import { IconButton } from '../uiComponents';

interface Props {
  className?: string;
  line: RouteLine;
}

export const LineTableRow = ({ className, line }: Props): JSX.Element => {
  const { t } = useTranslation();
  const { showRoutesOnModal } = useShowRoutesOnModal();

  const showLineRoutes = () => {
    const lineRouteIds = line.line_routes?.map((item) => item.route_id);
    showRoutesOnModal(lineRouteIds);
  };

  return (
    <tbody>
      <tr className={`border ${className}`}>
        <td className="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold">
          <Link to={routes[Path.lineDetails].getLink(line.line_id)}>
            <Row>
              <Column className="w-1/2">
                <p className="text-3xl">{line.label}</p>
                <p className="text-lg">{line.name_i18n}</p>
              </Column>
              <Column className="w-1/2 text-right">
                <p className="text-lg font-bold">
                  {t('validity.validDuring', {
                    startDate: mapToShortDate(line.validity_start || MIN_DATE),
                    endDate: mapToShortDate(line.validity_end || MAX_DATE),
                  })}
                </p>
                <p className="text-lg">
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </Column>
            </Row>
          </Link>
        </td>
        <td className="w-20 border">
          <IconButton
            className="h-full w-full"
            onClick={showLineRoutes}
            icon={<MdPinDrop className="text-5xl text-tweaked-brand" />}
            testId="LineTableRow::showLineRoutes"
          />
        </td>
      </tr>
    </tbody>
  );
};
