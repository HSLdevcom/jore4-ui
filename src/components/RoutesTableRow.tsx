import qs from 'qs';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { MdPinDrop } from 'react-icons/md';
import { Link } from 'react-router-dom';
import { RouteRoute } from '../generated/graphql';
import { Column, Row } from '../layoutComponents';
import { Path, routes } from '../routes'; // eslint-disable-line import/no-cycle
import { mapToShortDate, MAX_DATE, MIN_DATE } from '../time';

interface Props {
  className?: string;
  route: Partial<RouteRoute>;
}

export const RoutesTableRow = ({ className, route }: Props): JSX.Element => {
  const { t } = useTranslation();
  const mapUrl = routes[Path.map].getLink();
  const openInMapUrl = `${mapUrl}?${qs.stringify({
    routeId: route.route_id,
  })}`;
  return (
    <tbody>
      <tr className={`border ${className}`}>
        <td className="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold">
          <Link to={routes[Path.lineDetails].getLink(route.on_line_id)}>
            <Row>
              <Column className="w-1/2">
                <p className="text-3xl">{route.label}</p>
                <p className="text-lg">{route.description_i18n}</p>
              </Column>
              <Column className="w-1/2 text-right">
                <p className="text-lg font-bold">
                  {t('validity.validDuring', {
                    startDate: mapToShortDate(route.validity_start || MIN_DATE),
                    endDate: mapToShortDate(route.validity_end || MAX_DATE),
                  })}
                </p>
                <p className="text-lg">
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </Column>
            </Row>
          </Link>
        </td>
        <td className="border">
          <Link to={openInMapUrl} className="flex justify-center py-3">
            <MdPinDrop className="text-center text-5xl text-tweaked-brand" />
          </Link>
        </td>
      </tr>
    </tbody>
  );
};
