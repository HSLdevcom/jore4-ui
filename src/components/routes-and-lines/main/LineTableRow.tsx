import { useTranslation } from 'react-i18next';
import { MdPinDrop } from 'react-icons/md';
import { Link } from 'react-router-dom';
import { RouteLine } from '../../../generated/graphql';
import { useShowRoutesOnModal } from '../../../hooks';
import { Column, Row } from '../../../layoutComponents';
import { Path, routeDetails } from '../../../router/routeDetails';
import { mapToShortDate, MAX_DATE, MIN_DATE } from '../../../time';
import { IconButton } from '../../../uiComponents';

interface Props {
  className?: string;
  line: RouteLine;
}

export const LineTableRow = ({ className, line }: Props): JSX.Element => {
  const { t } = useTranslation();
  const { showRoutesOnModal } = useShowRoutesOnModal();

  const showLineRoutes = () => {
    const lineRouteIds = line.line_routes?.map((item) => item.route_id);

    showRoutesOnModal(
      lineRouteIds,
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      line.validity_start!,
      line.validity_end,
    );
  };

  // FIXME: quick and dirty hack for displaying the name that comes from the jore3 import
  const name = line.name_i18n?.startsWith('{"fi_FI"')
    ? JSON.parse(line.name_i18n).fi_FI
    : line.name_i18n;

  return (
    <tbody>
      <tr className={`border ${className}`}>
        <td className="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold">
          <Link to={routeDetails[Path.lineDetails].getLink(line.line_id)}>
            <Row>
              <Column className="w-1/2">
                <p className="text-3xl">{line.label}</p>
                <p className="text-lg">{name}</p>
              </Column>
              <Column className="w-1/2 text-right">
                <p className="text-lg font-bold">
                  {t('validity.validDuring', {
                    startDate: mapToShortDate(line.validity_start || MIN_DATE),
                    endDate: mapToShortDate(line.validity_end || MAX_DATE),
                  })}
                </p>
                <p className="text-lg">
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </Column>
            </Row>
          </Link>
        </td>
        <td className="w-20 border">
          <IconButton
            className="h-full w-full"
            onClick={showLineRoutes}
            icon={<MdPinDrop className="text-5xl text-tweaked-brand" />}
            testId="LineTableRow::showLineRoutes"
          />
        </td>
      </tr>
    </tbody>
  );
};
