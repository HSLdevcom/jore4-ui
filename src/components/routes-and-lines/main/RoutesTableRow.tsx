import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { RouteRoute } from '../../../generated/graphql';
import { useShowRoutesOnModal } from '../../../hooks';
import { Column, Row } from '../../../layoutComponents';
import { Path, routeDetails } from '../../../router/routeDetails';
import { mapToShortDate, MAX_DATE, MIN_DATE } from '../../../time';
import { LocatorButton } from '../../../uiComponents';

interface Props {
  className?: string;
  route: RouteRoute;
}

export const RoutesTableRow = ({ className, route }: Props): JSX.Element => {
  const { t } = useTranslation();
  const { showRoutesOnModal } = useShowRoutesOnModal();

  return (
    <tbody>
      <tr className={`border ${className}`}>
        <td className="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold">
          <Link to={routeDetails[Path.lineDetails].getLink(route.on_line_id)}>
            <Row>
              <Column className="w-1/2">
                <p className="text-3xl">{route.label}</p>
                <p className="text-lg">{route.name_i18n?.fi_FI}</p>
              </Column>
              <Column className="w-1/2 text-right">
                <p className="text-lg font-bold">
                  {t('validity.validDuring', {
                    startDate: mapToShortDate(route.validity_start || MIN_DATE),
                    endDate: mapToShortDate(route.validity_end || MAX_DATE),
                  })}
                </p>
                <p className="text-lg">
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </Column>
            </Row>
          </Link>
        </td>
        <td className="w-20 border text-center">
          <LocatorButton
            onClick={() =>
              showRoutesOnModal(
                [route.route_id],
                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                route.validity_start!,
                route.validity_end,
              )
            }
            disabled={
              !route.route_shape /* some routes imported from jore3 are missing the geometry */
            }
            testId="RoutesTableRow::showRoute"
          />
        </td>
      </tr>
    </tbody>
  );
};
