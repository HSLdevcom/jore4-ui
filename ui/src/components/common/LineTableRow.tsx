import { gql } from '@apollo/client';
import { LineTableRowFragment } from '../../generated/graphql';
import {
  useAppDispatch,
  useAppSelector,
  useShowRoutesOnModal,
} from '../../hooks';
import {
  deselectRouteUniqueLabelsAction,
  selectExport,
  selectRouteUniqueLabelsAction,
} from '../../redux';
import { routeHasTimetables } from '../../utils/route';
import {
  RouteLineTableRow,
  RouteLineTableRowVariant,
} from './RouteLineTableRow';

interface Props {
  className?: string;
  line: LineTableRowFragment;
  isSelectable?: boolean;
  rowVariant: RouteLineTableRowVariant;
}

const GQL_LINE_TABLE_ROW = gql`
  fragment line_table_row on route_line {
    name_i18n
    short_name_i18n
    priority
    ...line_map_params
    line_routes {
      ...route_map_params
      unique_label
      route_journey_patterns {
        journey_pattern_id
        journey_pattern_refs {
          journey_pattern_ref_id
          vehicle_journeys {
            vehicle_journey_id
          }
        }
      }
    }
  }
`;

/**
 * Reusable component LineTableRow for list views. This component requires
 * the line information (LineTableRowFragment) and linkTo parameter to
 * determine where we navigate after clicking this row.
 */
export const LineTableRow = ({
  className = '',
  line,
  isSelectable = false,
  rowVariant,
}: Props): JSX.Element => {
  const { showRoutesOnMapByLineLabel } = useShowRoutesOnModal();
  const dispatch = useAppDispatch();
  const { selectedRouteUniqueLabels } = useAppSelector(selectExport);

  const showLineRoutes = () => {
    showRoutesOnMapByLineLabel(line);
  };

  const onSelectChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
    const selected = event.target.checked;
    const selectAction = selected
      ? selectRouteUniqueLabelsAction
      : deselectRouteUniqueLabelsAction;

    dispatch(selectAction(line.line_routes.map((route) => route.label)));
  };

  const hasRoutes = line.line_routes?.length > 0;

  // Entire line is selected for export if it has routes and each of those is selected.
  // Selecting only subset of routes shouldn't be possible, as user can select routes
  // from line list or from route list, not from both simultaneously.
  const isSelected =
    hasRoutes &&
    line.line_routes.every((route) =>
      selectedRouteUniqueLabels.includes(route.label),
    );

  // Check if any of the line's route has timetables existing
  const hasTimetables = line.line_routes.some(routeHasTimetables);

  return (
    <RouteLineTableRow
      rowItem={line}
      rowVariant={rowVariant}
      hasTimetables={hasTimetables}
      onLocatorButtonClick={showLineRoutes}
      locatorButtonTestId="LineTableRow::showLineRoutes"
      className={className}
      lineId={line.line_id}
      onSelectChanged={isSelectable ? onSelectChanged : undefined}
      isSelected={isSelected}
      selectionDisabled={!hasRoutes}
    />
  );
};
