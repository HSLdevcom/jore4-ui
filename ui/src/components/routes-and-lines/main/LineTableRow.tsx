import { gql } from '@apollo/client';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { LineTableRowFragment } from '../../../generated/graphql';
import { useAlertsAndHighLights, useShowRoutesOnModal } from '../../../hooks';
import { Column, Row } from '../../../layoutComponents';
import { Path, routeDetails } from '../../../router/routeDetails';
import { mapToShortDate, MAX_DATE, MIN_DATE } from '../../../time';
import { LocatorButton } from '../../../uiComponents';

interface Props {
  className?: string;
  line: LineTableRowFragment;
}

const GQL_LINE_TABLE_ROW = gql`
  fragment line_table_row on route_line {
    ...line_all_fields
    line_routes {
      route_id
    }
  }
`;

export const LineTableRow = ({ className = '', line }: Props): JSX.Element => {
  const { t } = useTranslation();
  const { showRoutesOnModal } = useShowRoutesOnModal();

  const showLineRoutes = () => {
    const lineRouteIds = line.line_routes?.map((item) => item.route_id);

    showRoutesOnModal(
      lineRouteIds,
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      line.validity_start!,
      line.validity_end,
    );
  };

  const { getAlertLevel, getAlertStyle } = useAlertsAndHighLights();
  const alertStyle = getAlertStyle(getAlertLevel(line));

  return (
    <tbody>
      <tr className={`border border-light-grey ${className}`}>
        <td className={`${alertStyle.listItemBorder || ''} w-20 p-4 align-top`}>
          {alertStyle.icon && <i className={`${alertStyle.icon} text-3xl`} />}
        </td>
        <td className="py-4">
          <Link to={routeDetails[Path.lineDetails].getLink(line.line_id)}>
            <Row className="items-center">
              <Column className="w-1/2 font-bold">
                <p className="text-3xl">{line.label}</p>
                <p>{line.name_i18n.fi_FI}</p>
              </Column>
              <Column className="w-1/2 text-right">
                <p className="font-bold">
                  {t('validity.validDuring', {
                    startDate: mapToShortDate(line.validity_start || MIN_DATE),
                    endDate: mapToShortDate(line.validity_end || MAX_DATE),
                  })}
                </p>
                <p>!Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija</p>
              </Column>
            </Row>
          </Link>
        </td>
        <td className="w-20 p-6 align-middle">
          <LocatorButton
            onClick={showLineRoutes}
            testId="LineTableRow::showLineRoutes"
          />
        </td>
      </tr>
    </tbody>
  );
};
