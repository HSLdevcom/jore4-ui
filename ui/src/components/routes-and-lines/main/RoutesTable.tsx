import React from 'react';

interface Props {
  className?: string;
  testId?: string;
}

export const RoutesTable: React.FC<Props> = ({
  className,
  testId,
  children,
}) => {
  return (
    // setting a fake "height: 1px" so that "height: 100%" would work for the table cells
    <table className={`h-1 w-full ${className}`} data-testid={testId}>
      {children}
    </table>
  );
};
