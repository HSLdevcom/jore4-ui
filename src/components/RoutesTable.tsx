import React from 'react';
import { RouteRoute } from '../generated/graphql';
import { RoutesTableRow } from './RoutesTableRow'; // eslint-disable-line import/no-cycle

interface Props {
  className?: string;
  routes: RouteRoute[];
  testId?: string;
}

export const RoutesTable = ({
  className,
  routes,
  testId,
}: Props): JSX.Element => {
  return (
    <table className={`w-full ${className}`} data-testid={testId}>
      <tbody>
        {routes.map((item: RouteRoute) => (
          <RoutesTableRow key={item.route_id} route={item} />
        ))}
      </tbody>
    </table>
  );
};
