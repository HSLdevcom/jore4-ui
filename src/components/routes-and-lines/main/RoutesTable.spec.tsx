import React from 'react';
import {
  ListChangingRoutesQuery,
  RouteDirectionEnum,
  RouteLine,
  RouteRoute,
  useListOwnLinesQuery,
} from '../../../generated/graphql';
import {
  GqlQueryResult,
  mapListChangingRoutesResult,
  mapListOwnLinesResult,
} from '../../../graphql';
import {
  buildLocalizedString,
  buildRoute,
  render,
} from '../../../utils/test-utils';
import { LineTableRow } from './LineTableRow';
import { RoutesTable } from './RoutesTable';
import { RoutesTableRow } from './RoutesTableRow';

describe(`<${RoutesTable.name} />`, () => {
  const testId = 'routesTable1';
  const route1Id = '03d55414-e5cf-4cce-9faf-d86ccb7e5f98';
  const lineId = '101f800c-39ed-4d85-8ece-187cd9fe1c5e';
  const stop1Id = 'f8eace87-7901-4438-bfee-bb6f24f1c4c4';
  const stop2Id = 'e3528755-711f-4e4f-9461-7931a2c4bc6d';

  // this response is copy-pasted from the actual graphql response
  const routesResponseMock: GqlQueryResult<ListChangingRoutesQuery> = {
    data: {
      route_route: [
        {
          __typename: 'route_route',
          ...buildRoute('1'),
          route_id: route1Id,
          on_line_id: lineId,
          priority: 10,
          direction: RouteDirectionEnum.Outbound,
          route_line: {
            line_id: lineId,
            label: '65',
            name_i18n: buildLocalizedString(
              'Rautatientori - Kätilöopisto - Veräjälaakso',
            ),
            short_name_i18n: buildLocalizedString(
              'Rautatientori - Veräjälaakso',
            ),
            __typename: 'route_line',
          },
          starts_from_scheduled_stop_point_id: stop1Id,
          starts_from_scheduled_stop_point: {
            scheduled_stop_point_id: stop1Id,
            label: 'pysäkki B',
            __typename: 'service_pattern_scheduled_stop_point',
          },
          ends_at_scheduled_stop_point_id: stop2Id,
          ends_at_scheduled_stop_point: {
            scheduled_stop_point_id: stop2Id,
            label: 'pysäkki A',
            __typename: 'service_pattern_scheduled_stop_point',
          },
        },
      ],
    },
  };

  test('Renders the table with route data', async () => {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const routes = mapListChangingRoutesResult(routesResponseMock)!;
    const { asFragment } = render(
      <RoutesTable testId={testId}>
        {routes.map((item: RouteRoute) => (
          <RoutesTableRow key={item.route_id} route={item} />
        ))}
      </RoutesTable>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // this response is copy-pasted from the actual graphql response
  const linesResponseMock = {
    data: {
      route_line: [
        {
          line_id: '101f800c-39ed-4d85-8ece-187cd9fe1c5e',
          label: '65',
          name_i18n: buildLocalizedString('Rautatientori - Veräjälaakso'),
          short_name_i18n: buildLocalizedString('Rautatientori - Veräjälaakso'),
          __typename: 'route_line',
        },
        {
          line_id: '9058c328-efdd-412c-9b2b-37b0f6a2c6fb',
          label: '71',
          name_i18n: buildLocalizedString('Rautatientori - Malmi as.'),
          short_name_i18n: buildLocalizedString('Rautatientori - Malmi as.'),
          __typename: 'route_line',
        },
        {
          line_id: 'bbd1cb29-74c3-4fa1-ac86-918d7fa96fe2',
          label: '785K',
          name_i18n: buildLocalizedString('Rautatientori - Nikkilä'),
          short_name_i18n: buildLocalizedString('Rautatientori - Nikkilä'),
          __typename: 'route_line',
        },
        {
          line_id: 'db748c5c-42e3-429f-baa0-e0db227dc2c8',
          label: '1234',
          name_i18n: buildLocalizedString('Erottaja - Arkkadiankatu'),
          short_name_i18n: buildLocalizedString('Erottaja - Arkkadiankatu'),
          __typename: 'route_line',
        },
      ],
    },
  } as ReturnType<typeof useListOwnLinesQuery>;

  test('Renders the table with line data', async () => {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const lines = mapListOwnLinesResult(linesResponseMock)!;
    const { asFragment } = render(
      <RoutesTable testId={testId}>
        {lines.map((item: RouteLine) => (
          <LineTableRow key={item.line_id} line={item} />
        ))}
      </RoutesTable>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
