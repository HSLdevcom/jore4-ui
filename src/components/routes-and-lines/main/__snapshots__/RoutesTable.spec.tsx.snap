// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoutesTable /> Renders the table with line data 1`] = `
<DocumentFragment>
  <table
    class="h-1 w-full undefined"
    data-testid="routesTable1"
  >
    <tbody>
      <tr
        class="border undefined"
      >
        <td
          class="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold"
        >
          <a
            href="/lines/101f800c-39ed-4d85-8ece-187cd9fe1c5e"
          >
            <div
              class="flex flex-row undefined"
            >
              <div
                class="flex flex-col w-1/2"
              >
                <p
                  class="text-3xl"
                >
                  65
                </p>
                <p
                  class="text-lg"
                >
                  Rautatientori - Veräjälaakso
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="text-lg font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p
                  class="text-lg"
                >
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-20 border"
        >
          <button
            class="text-center h-full w-full"
            data-testid="LineTableRow::showLineRoutes"
            type="button"
          >
            <svg
              class="text-5xl text-tweaked-brand inline text-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        class="border undefined"
      >
        <td
          class="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold"
        >
          <a
            href="/lines/9058c328-efdd-412c-9b2b-37b0f6a2c6fb"
          >
            <div
              class="flex flex-row undefined"
            >
              <div
                class="flex flex-col w-1/2"
              >
                <p
                  class="text-3xl"
                >
                  71
                </p>
                <p
                  class="text-lg"
                >
                  Rautatientori - Malmi as.
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="text-lg font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p
                  class="text-lg"
                >
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-20 border"
        >
          <button
            class="text-center h-full w-full"
            data-testid="LineTableRow::showLineRoutes"
            type="button"
          >
            <svg
              class="text-5xl text-tweaked-brand inline text-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        class="border undefined"
      >
        <td
          class="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold"
        >
          <a
            href="/lines/bbd1cb29-74c3-4fa1-ac86-918d7fa96fe2"
          >
            <div
              class="flex flex-row undefined"
            >
              <div
                class="flex flex-col w-1/2"
              >
                <p
                  class="text-3xl"
                >
                  785K
                </p>
                <p
                  class="text-lg"
                >
                  Rautatientori - Nikkilä
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="text-lg font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p
                  class="text-lg"
                >
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-20 border"
        >
          <button
            class="text-center h-full w-full"
            data-testid="LineTableRow::showLineRoutes"
            type="button"
          >
            <svg
              class="text-5xl text-tweaked-brand inline text-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        class="border undefined"
      >
        <td
          class="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold"
        >
          <a
            href="/lines/db748c5c-42e3-429f-baa0-e0db227dc2c8"
          >
            <div
              class="flex flex-row undefined"
            >
              <div
                class="flex flex-col w-1/2"
              >
                <p
                  class="text-3xl"
                >
                  1234
                </p>
                <p
                  class="text-lg"
                >
                  Erottaja - Arkkadiankatu
                </p>
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="text-lg font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p
                  class="text-lg"
                >
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-20 border"
        >
          <button
            class="text-center h-full w-full"
            data-testid="LineTableRow::showLineRoutes"
            type="button"
          >
            <svg
              class="text-5xl text-tweaked-brand inline text-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<RoutesTable /> Renders the table with route data 1`] = `
<DocumentFragment>
  <table
    class="h-1 w-full undefined"
    data-testid="routesTable1"
  >
    <tbody>
      <tr
        class="border undefined"
      >
        <td
          class="border-l-8 border-hsl-dark-green py-4 pl-16 pr-4 font-bold"
        >
          <a
            href="/lines/101f800c-39ed-4d85-8ece-187cd9fe1c5e"
          >
            <div
              class="flex flex-row undefined"
            >
              <div
                class="flex flex-col w-1/2"
              >
                <p
                  class="text-3xl"
                />
                <p
                  class="text-lg"
                />
              </div>
              <div
                class="flex flex-col w-1/2 text-right"
              >
                <p
                  class="text-lg font-bold"
                >
                  Voimassa 1.1.1970 - 31.12.2050
                </p>
                <p
                  class="text-lg"
                >
                  !Muokattu dd.mm.yyyy hh:mm | S. Suunnittelija
                </p>
              </div>
            </div>
          </a>
        </td>
        <td
          class="w-20 border"
        >
          <button
            class="text-center h-full w-full"
            data-testid="RoutesTableRow::showRoute"
            type="button"
          >
            <svg
              class="text-5xl text-tweaked-brand inline text-center"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
